packages/catalog-readmodel-writer/src/consumerServiceV1.ts

```diff
@@ -1,11 +1,39 @@
 import { match } from "ts-pattern";
 import { EServiceCollection } from "pagopa-interop-commons";
-import { EServiceEventEnvelopeV1 } from "pagopa-interop-models";
 import {
-  fromDescriptorV1,
-  fromDocumentV1,
-  fromEServiceV1,
-} from "./model/converterV1.js";
+  EServiceEventEnvelopeV1,
+  EServiceV1,
+  EServiceLegacy,
+  EServiceDocumentV1,
+  EServiceDescriptorV1,
+  DescriptorLegacy,
+  DocumentLegacy,
+} from "pagopa-interop-models";
+import {
+  fromDescriptorV1Legacy,
+  fromDocumentV1Legacy,
+  fromEServiceV1Legacy,
+} from "./model/legacy/eserviceLegacyConverter.js";
+
+const toEServiceReadModel = (
+  version: number,
+  eservice?: EServiceV1
+): { data: EServiceLegacy | undefined; metadata: { version: number } } => ({
+  data: eservice ? fromEServiceV1Legacy(eservice) : undefined,
```

Comment:
These new `from...V1Legacy` / `from...V2Legacy` methods are misleading to me: `from...V*` methods are used to deserialize the data from the event payload and should be unrelated to the interaction with the readmodel. 

These new `from` methods make it look like the problem was in deserialization from the events payload, not in serialization to the readmodel. I would leave the `from...` methods as they were and, here in the `to...ReadModel` methods, change the date serialization to write ISO strings in the DB, to clarify that the fix is just for serialization to the readmodel.

(same for all the new `from...Legacy` methods modified in the PR)